client/state/domains/suggestions/actions.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	DOMAINS_SUGGESTIONS_RECEIVE,
+	DOMAINS_SUGGESTIONS_REQUEST,
+	DOMAINS_SUGGESTIONS_REQUEST_FAILURE,
+	DOMAINS_SUGGESTIONS_REQUEST_SUCCESS
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that a domains suggestion object
+ * has been received.
+ *
+ * @param   {Array}    suggestions              domain suggestions
+ * @param   {Object}   queryObject              domain suggestions queryObject
+ * @returns {Object}   Action object
+ */
+export function receiveDomainsSuggestions( suggestions, queryObject ) {
+	return {
+		type: DOMAINS_SUGGESTIONS_RECEIVE,
+		queryObject,
+		suggestions
```

Comment:
Using an assembler function was useful to omit extraneous data from old endpoints and handle type inconsistencies in the API, but I agree that we can avoid using them. Snake → camel was just an added benefit as far as I remember - we could create a general solution function for this using _.camelCase if we want.